azure: minio-java test fails on azure gateway

## Expected Behavior
Minio-Java functional tests work on azure gateway 

## Current Behavior
Minio-Java Functional Tests fails on azure backend. 

```
$ ./gradlew runFunctionalTest
:api:compileJava UP-TO-DATE
:api:processResources UP-TO-DATE
:api:classes UP-TO-DATE
:api:jar UP-TO-DATE
:functional:compileJava UP-TO-DATE
:functional:processResources UP-TO-DATE
:functional:classes UP-TO-DATE
:functional:runFunctionalTest
Test: makeBucket(String bucketName)
Test: listBuckets()
Test: bucketExists(String bucketName)
Test: removeBucket(String bucketName)
Test: putObject(String bucketName, String objectName, String filename)
Test: multipart: putObject(String bucketName, String objectName, String filename)
Test: multipart resume: putObject(String bucketName, String objectName, String filename)
Test: putObject(String bucketName, String objectName, String contentType, long size, InputStream body)
Test: putObject(String bucketName, String objectName, String filename, String contentType)
Test: objectName with path segments: putObject(String bucketName, String objectName, String filename)
Test: putObject(String bucketName, String objectName, InputStream body, String contentType)
Test: multipart: putObject(String bucketName, String objectName, InputStream body, String contentType)
java.net.SocketException: Connection reset
        at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115)
        at java.net.SocketOutputStream.write(SocketOutputStream.java:155)
        at okio.Okio$1.write(Okio.java:78)
        at okio.AsyncTimeout$1.write(AsyncTimeout.java:179)
        at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171)
        at okio.RealBufferedSink.write(RealBufferedSink.java:41)
        at okhttp3.internal.http1.Http1Codec$FixedLengthSink.write(Http1Codec.java:286)
        at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171)
        at okio.RealBufferedSink.write(RealBufferedSink.java:109)
        at io.minio.MinioClient$3.writeTo(MinioClient.java:871)
        at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:62)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
        at okhttp3.RealCall.execute(RealCall.java:69)
        at io.minio.MinioClient.execute(MinioClient.java:998)
        at io.minio.MinioClient.executePut(MinioClient.java:1266)
        at io.minio.MinioClient.executePut(MinioClient.java:1288)
        at io.minio.MinioClient.putObject(MinioClient.java:3480)
        at io.minio.MinioClient.putObject(MinioClient.java:3590)
        at io.minio.MinioClient.putObject(MinioClient.java:3175)
        at FunctionalTest.putObject_test8(FunctionalTest.java:291)
        at FunctionalTest.runTests(FunctionalTest.java:1601)
        at FunctionalTest.main(FunctionalTest.java:1704)
:functional:runFunctionalTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':functional:runFunctionalTest'.
> Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 255

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 59.525 secs

```

## Possible Solution
Not known

## Steps to Reproduce (for bugs)
- Build Minio server from current master
- Run it in `azure` gateway mode. 
- Point minio-java functional tests to your Minio instance and run with `./gradlew runFunctionalTest`

--- a/cmd/api-errors.go
+++ b/cmd/api-errors.go
@@ -725,6 +725,8 @@ func toAPIErrorCode(err error) (apiErr APIErrorCode) {
 		apiErr = ErrNotImplemented
 	case PolicyNotFound:
 		apiErr = ErrNoSuchBucketPolicy
+	case PartTooBig:
+		apiErr = ErrEntityTooLarge
 	default:
 		apiErr = ErrInternalError
 	}

--- a/cmd/gateway-azure.go
+++ b/cmd/gateway-azure.go
@@ -137,6 +137,8 @@ func azureToObjectError(err error, params ...string) error {
 		err = BucketExists{Bucket: bucket}
 	case "InvalidResourceName":
 		err = BucketNameInvalid{Bucket: bucket}
+	case "RequestBodyTooLarge":
+		err = PartTooBig{}
 	default:
 		switch azureErr.StatusCode {
 		case http.StatusNotFound:

--- a/cmd/object-api-errors.go
+++ b/cmd/object-api-errors.go
@@ -343,6 +343,13 @@ func (e PartTooSmall) Error() string {
 	return fmt.Sprintf("Part size for %d should be atleast 5MB", e.PartNumber)
 }
 
+// PartTooBig returned if size of part is bigger than the allowed limit.
+type PartTooBig struct{}
+
+func (e PartTooBig) Error() string {
+	return "Part size bigger than the allowed limit"
+}
+
 // NotImplemented If a feature is not implemented
 type NotImplemented struct{}
 

